packages/store/src/Storage.sol

```diff
@@ -88,107 +138,114 @@ library Storage {
     uint256 storagePointer,
     uint256 length,
     uint256 offset
-  ) internal view returns (bytes memory) {
-    Buffer buffer = Buffer_.allocate(uint128(length));
-    load(storagePointer, length, offset, buffer);
-    return buffer.toBytes();
+  ) internal view returns (bytes memory result) {
+    // TODO this will probably use less gas via manual memory allocation
+    result = new bytes(length);
+    uint256 memoryPointer;
+    assembly {
+      memoryPointer := add(result, 0x20)
+    }
+    load(storagePointer, length, offset, memoryPointer);
+    return result;
   }
 
   /**
-   * @notice Append raw bytes from storage at the given storagePointer, offset, and length to the given buffer
+   * @notice Load into the buffer from storage at the given storagePointer, offset, and length
    */
   function load(
     uint256 storagePointer,
     uint256 length,
     uint256 offset,
     Buffer buffer
   ) internal view {
-    // Support offsets that are greater than 32 bytes by incrementing the storagePointer and decrementing the offset
-    storagePointer += offset / 32;
-    offset %= 32;
-
-    uint256 numWords = Utils.divCeil(length + offset, 32);
-    uint256 _lengthToload;
-
-    for (uint256 i; i < numWords; i++) {
-      // If this is the first word, and there is an offset, apply a mask to beginning (and possibly the end if length + offset is less than 32)
-      if ((i == 0 && offset > 0)) {
-        _lengthToload = length + offset > 32 ? 32 - offset : length; // the number of bytes to load
-        buffer.appendUnchecked(
-          _loadPartialWord(
-            storagePointer, // the slot to start loading from
-            _lengthToload,
-            offset // the offset in bytes to start loading from
-          ),
-          uint128(_lengthToload)
-        );
-
-        // If this is the last word, and there is a partial word, apply a mask to the end
-      } else if (i == numWords - 1 && (length + offset) % 32 > 0) {
-        _lengthToload = (length + offset) % 32; //  the relevant length of the trailing word
-        buffer.appendUnchecked(
-          _loadPartialWord(
-            storagePointer + i, // the word to load from
-            _lengthToload,
-            0 // the offset in bytes to start loading from
-          ),
-          uint128(_lengthToload)
-        );
-
-        // Else, just load the word
-      } else {
-        buffer.appendUnchecked(_loadWord(storagePointer + i), 32);
-      }
-    }
-  }
+    uint256 bufferLength = buffer.length();
+    load(storagePointer, length, offset, buffer.ptr() + bufferLength);
```

Comment:
why are we adding the buffer's length to the buffer's pointer to get the memory pointer?